packages/back-end/src/api/sdk-connections/validations.ts

```diff
@@ -102,14 +120,30 @@ export const validatePayload = async (
 
   capabilityParams.forEach(([capability, param]) => {
     if (payload[param] && !capabilities.includes(capability))
-      throw new Error(
-        `SDK version ${sdkVersion} doesn not support ${capability}`
-      );
+      if (latestCapabilities.includes(capability))
+        throw new Error(
+          `You need to ugrade to version ${latestSdkVersion} to support ${capability}`
+        );
+      else
+        throw new Error(
+          `SDK version ${sdkVersion} doesn not support ${capability}`
+        );
   });
 
   premiumFeatures.forEach(([feature, param]) => {
-    if (payload[param] && !context.hasPremiumFeature(feature))
+    if (!payload[param]) return;
+
+    if (!context.hasPremiumFeature(feature))
       throw new Error(`Feature ${feature} requires premium subscription!`);
+
+    const extraCheck = premiumExtraChecks[feature];
+
+    if (extraCheck === undefined) return;
+
+    if (!extraCheck.check)
+      throw new Error(
+        `Feature ${feature} is not available: ${extraCheck.reason}`
+      );
```

Comment:
I think there's some misunderstandings here.

On self-hosted, the proxy settings are always allowed.  On cloud, the proxy settings are only allowed if the user has the `cloud-proxy` premium feature.

Reply:
Gotcha